jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -5307,4 +5307,10 @@ public IntStream codePoints() {
 		return CharSequence.super.codePoints();
 	}
 /*[ENDIF]*/	
+
+/*[IF Java10]*/
+	byte[] value() {
+		return java.util.Arrays.copyOf(value, lengthInternal()*2);
```

Comment:
This solution seems incomplete. lengthInternal()*2 is only appropriate when the string is not compressed.

Reply:
Is it always necessary to copy `value`? I expect the intent is to get the value without copying so doing a copy could impose an additional performance cost.

Reply:
Please document the reasoning for the implementation in the code and not just the commit comment.

Reply:
and btw I expect the reason that `value` can be longer than necessary is the substring() implementation since by default enableSharingInSubstringWhenOffsetIsZero is true.

Reply:
Fixed, pls have another look.